src/main/java/uk/ac/ebi/atlas/search/metadata/MetadataSearchDao.java

```diff
@@ -0,0 +1,92 @@
+package uk.ac.ebi.atlas.search.metadata;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.download.FileDownloadController;
+import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
+import uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy;
+import uk.ac.ebi.atlas.solr.cloud.search.SolrQueryBuilder;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CELL_ID;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CHARACTERISTIC_NAME;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.CHARACTERISTIC_VALUE;
+import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.EXPERIMENT_ACCESSION;
+
+@Component
+public class MetadataSearchDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileDownloadController.class);
+    private SingleCellAnalyticsCollectionProxy singleCellAnalyticsCollectionProxy;
+
+    public MetadataSearchDao(SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory) {
+        this.singleCellAnalyticsCollectionProxy =
+                solrCloudCollectionProxyFactory.create(SingleCellAnalyticsCollectionProxy.class);
+    }
+
+
+    private String getInferredCellTypeForCellId(String experimentAccession, String cellId) {
+        var characteristicFields = "inferred_cell_type";
```

Comment:
I believe this should be renamed to `characteristicField` as it is just single field name.

Reply:
This variable is just declared and used once. I think inlinining `"inferred_cell_type"` is better (fewer indirections when you read the code).

Always be wary of one-use variables.